Achilles/World.pde

```diff
@@ -0,0 +1,50 @@
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * World represents the entire running world of the game. It should expose functionality
+ * to the base Achilles app so that it can respond to input, update itself, and draw itself
+ */
+class World {
+  private Set<Entity> entities = new HashSet<Entity>();
+  private Set<Entity> entering = new HashSet<Entity>();
+  private List<Laser> lasers = new ArrayList<Laser>();
+  
+  public World() {
+    lasers.add(new Laser(0, height));
+    lasers.add(new Laser(width / 2, height));
+    lasers.add(new Laser(width, height));
+    entering.addAll(lasers);
+  }
+  
+  public void addEntities(Entity... newEntities) {
+    entering.addAll(Arrays.asList(newEntities)); 
```

Comment:
A couple things: if you've never seen a method signature with the ellipsis in it, that's called a [variadic argument](https://en.wikipedia.org/wiki/Variadic_function). It just means that you can pass the function any number of entities to add. It's essentially shorthand for `public void addEntities(Entity[] entities)` but with the bonus that you can now call it with comma-separated arguments like `world.addEntities(entityOne, entityTwo)`

Second, I use the `entering` array as opposed to adding directly to `entities` because `addEntities` could potentially be called at any point during a frame, and there could be some weird logic that happens if the `entities` array is modified *while* it's being iterated over during `update()` or `render()` so instead I buffer the new entities until the beginning of each frame when they're transferred over.